client/components/domains/register-domain-step/index.jsx

```diff
@@ -263,6 +271,25 @@ class RegisterDomainStep extends Component {
 			this.setState( this.getState( nextProps ) );
 			nextProps.suggestion && this.onSearch( nextProps.suggestion );
 		}
+
+		// Filter out the free wp.com subdomains to avoid doing another API request.
+		// Please note that it's intentional to be incomplete -- the complete version of this
+		// should be able to handle flag transition the other way around, i.e.
+		// when `includeWordPressDotCom` is first `false` and then transit to `true`. The
+		// same should also be ported to the dotblog subdomain flag. However, this code is likely
+		// temporary specific for the hiding free subdomain test, so it's not practical to implement
+		// the complete version for now.
+		if (
```

Comment:
Non blocking: This part LGTM, but I was curious about why this is required?

Reply:
Since we search for the suggested domains optimistically, there is an edge case where the experiment hasn't been loaded but the domain search request has already fired with `includeWordPressDotCom` as true. When that happen, `subdomainSearchResults` will contain the free subdomains so this chunk of code will filter them out. 

Does that make sense?

Reply:
Found that I missed to cover the other side of the request completion timing issue. I've submitted a fix as https://github.com/Automattic/wp-calypso/pull/82708/commits/4b08236ec0466b66d37670f7a2aa0da23cffa180